src/app/task.service.ts

```diff
@@ -24,12 +25,12 @@ export class TaskService {
   }
 
   getTasksInProgress(): Observable<Task[]> {
-    this.tasks = this.tasks.filter(t => t.status !== 1);
-    return of(this.tasks);
+    const taskInProgress = this.tasks.filter(t => t.status !== Status.DONE);
```

Comment:
Change variable name to tasksInProgress.
Change filter logic to - t.status === Status.IN_PRROGRESS (why would you do it the other way around? What if we add new status in the future. Then it would break your logic here)